core-bundle/src/Routing/AbstractPageRouteProvider.php

```diff
@@ -189,19 +211,22 @@ protected function compareRoutes(Route $a, Route $b, array $languages = null): i
 
     protected function convertLanguagesForSorting(array $languages): array
     {
-        foreach ($languages as &$language) {
+        $result = [];
+
+        foreach ($languages as $language) {
             $language = str_replace('_', '-', $language);
+            $result[] = $language;
 
             if (5 === \strlen($language)) {
                 $lng = substr($language, 0, 2);
 
                 // Append the language if only language plus dialect is given (see #430)
-                if (!\in_array($lng, $languages, true)) {
-                    $languages[] = $lng;
+                if (!\in_array($lng, $languages, true) && !\in_array($lng, $result, true)) {
```

Comment:
This checks `$result` on every single loop iteration. Wouldn't it be faster to have one `array_unique()` call after collecting them all?

Reply:
improved in f7b7c2e5da7e0abfa712960782a2da3e86bc3233